azure-devops/create-prdiff.ps1

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+[CmdletBinding(PositionalBinding = $False)]
+Param(
+    [Parameter(Mandatory = $True)]
+    [String]$DiffFile
+)
+
+Start-Process -FilePath 'git' -ArgumentList 'diff' `
+    -NoNewWindow -Wait `
+    -RedirectStandardOutput $DiffFile
+if (0 -ne (Get-Item -LiteralPath $DiffFile).Length) {
+    $msg = @(
+        'The formatting of the files in the repo was not what we expected.'
+        'Please access the diff from format.diff in the build artifacts,'
+        'and apply it with `git apply`.'
+        'Alternatively, you can run the `format` CMake target:'
+        '    cmake --build <builddir> --target format'
+        ''
+    )
+    Write-Host ($msg -join "`n")
+    Write-Host [string](Get-Content -LiteralPath $DiffFile)

```

Comment:
This isn't working well (https://dev.azure.com/vclibs/STL/_build/results?buildId=10878&view=logs&j=ce9c9df7-f6fa-59cf-0934-3339753bfc56&t=c6e68a6e-a0d3-5ff7-0941-3a725df1e075&l=19). `[string]` is emitted into the log, followed by the diff mashed onto one line. Perhaps `git` is emitting plain LF line endings that confuse powershell?

Reply:
Although I note that the diff file itself (https://dev.azure.com/vclibs/STL/_build/results?buildId=10878&view=artifacts&pathAsName=false&type=publishedArtifacts) has CRLF line endings. Maybe `Write-Host` itself is the problem.

Reply:
Consider using [Tee-Object](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/tee-object?view=powershell-7.2) on the output of `git diff` instead.

Reply:
Yeah, I'll see what I can do